client/src/views/calendar/calendar.ts

```diff
@@ -1,9 +1,77 @@
 import type { Observer } from '@/core/ui/observer';
+import styles from './calendar.module.scss';
 
 export default class Calendar implements Observer {
+  refactorData(state: StoreState): [number[][], number, number] {
+    const { records, categories } = state;
+    const recordArr = Array.from(Array(32), () => [0, 0, 0]);
+    let incomeTotal = 0;
+    let expenditureTotal = 0;
+
+    records.forEach((item) => {
+      const date = Number(String(item.date).split('-')[2]);
+      if (categories[item.categoryId - 1].type === 'income') {
+        recordArr[date][0] += item.value;
+        recordArr[date][2] += item.value;
+        incomeTotal += item.value;
+      } else {
+        recordArr[date][1] -= item.value;
+        recordArr[date][2] -= item.value;
+        expenditureTotal -= item.value;
+      }
+    });
+
+    return [recordArr, incomeTotal, expenditureTotal];
+  }
+
   template(state: StoreState): string {
+    const [recordArr, incomeTotal, expenditureTotal] = this.refactorData(state);
+
+    const yearNum = new Date().getFullYear();
+    const monthNum = new Date().getMonth();
+    const dayNum = new Date(yearNum, monthNum + 1, 0).getDate();
+    const frontBlankNum = new Date(yearNum, monthNum, 1).getDay();
+    const backBlankNum = 6 - new Date(yearNum, monthNum, dayNum).getDay();
+    const BlankArr = (num: number) => Array(num).fill(`<div class="${styles.block} ${styles.blank}"></div>`);
+    const dayArr = (num: number) =>
+      Array.from(Array(num), (_, index) => {
+        const idx = index + 1;
+        const income = recordArr[idx][0] === 0 ? '' : recordArr[idx][0];
+        const expenditure = recordArr[idx][1] === 0 ? '' : recordArr[idx][1];
+        let total: number | string = '';
+        if (income !== '' || expenditure !== '') [, , total] = recordArr[idx];
+        return `<div class="${styles.block}" name="${idx}">
```

Comment:
```suggestion
        return `<div class="${styles.block}" data-day="${idx}">
```

아까 말씀해 주실때 없어도 되기는 한다고 하셨는데 만약 남겨 둔다면 `data-` attribute로 변경하는게 좋을 것 같습니다.